core/model/modx/rest/modrestservice.class.php

```diff
@@ -242,8 +242,10 @@ protected function getController() {
      */
     public function iterateDirectories($pattern, $flags = 0) {
         $files = glob($pattern, $flags);
-        foreach (glob(dirname($pattern).'/*', GLOB_ONLYDIR|GLOB_NOSORT) as $dir) {
-            $files = array_merge($files, $this->iterateDirectories($dir.'/'.basename($pattern), $flags));
+        if ($dirs = glob(dirname($pattern).'/*', GLOB_ONLYDIR|GLOB_NOSORT)) {
```

Comment:
It's not very readable, not a good practice and is unnecessary terseness.

Please declare the variable before condition....
```php
$dirs = glob(dirname($pattern).'/*', GLOB_ONLYDIR|GLOB_NOSORT);
if ($dirs) {
```